### PR TITLE
Front-end updated with license detection

### PR DESCRIPTION
**Pull request by kawsarahamad**

### CODE REVIEW

File: src/controllers/user/user.controllers.js [hunk 1]
```diff
@@ -15,20 +22,50 @@ const getUserDashboard = async (req, res) => {
   }
 }
 
+// const addVehicle = async (req, res) => {
```
Comment (kawsarahamad) [inline src/controllers/user/user.controllers.js]: Remove your comments

File: src/controllers/user/user.controllers.js [hunk 2]
```diff
@@ -15,20 +22,50 @@ const getUserDashboard = async (req, res) => {
...
+// }
+
 const addVehicle = async (req, res) => {
-  const {licenseNumber, vehicleName} = req.body;
+  const { licenseNumber, vehicleName } = req.body;
   const _mail = req.user.email;
   try {
+    const vehicle = await Vehicle.findVehicleByLicenseNumber(licenseNumber);
+    if (vehicle != null) {
+      res.render('user/user.dashboard.ejs', {
```
Comment (Talha76) [inline src/controllers/user/user.controllers.js:51]: Does this flash work?

File: src/controllers/user/user.controllers.js [hunk 3]
```diff
@@ -15,20 +22,50 @@ const getUserDashboard = async (req, res) => {
...
+        error: 'Vehicle already exists', // Set an error message
```
Comment (Talha76) [inline src/controllers/user/user.controllers.js]: Does this way of flashing work?

File: src/routes/user/user.routes.js [hunk 4]
```diff
@@ -5,6 +5,6 @@ import userMiddlewares from '../../middlewares/user/user.middlewares.js';
 const router = Router();
 
 router.get('/dashboard', userMiddlewares.isloggedIn, userMiddlewares.emailVerificationMiddleware, userControllers.getUserDashboard);
-router.post('/dashboard', userMiddlewares.isloggedIn, userControllers.addVehicle);
+router.post('/dashboard/addVehicle', userMiddlewares.isloggedIn, userControllers.addVehicle);
```
Comment (Talha76) [inline src/routes/user/user.routes.js]: Don't use upper-casing in url-path (except tokens and others)

File: src/views/user/user.dashboard.ejs [hunk 5]
```diff
@@ -126,5 +123,17 @@
 <script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
 <script src="extract-lptext.js"></script>
 
+<!-- Your existing HTML content -->
```
Comment (Talha76) [inline src/views/user/user.dashboard.ejs]: Explain this part

**Review by Talha76 — CHANGES_REQUESTED** [review]
Change the requested parts.

File: src/routes/user/user.routes.js (hunk 4 above)
Comment (Shahrier-Al-Tanzim) [reply src/routes/user/user.routes.js]: so which one is preferable?
/add-vehicle or /addvehicle

File: src/controllers/user/user.controllers.js (hunk 3 above)
Comment (Shahrier-Al-Tanzim) [reply src/controllers/user/user.controllers.js]: Co-pilot gave this to me, so I think it should work, but it is not being detected in our code, so I removed it

File: src/controllers/user/user.controllers.js (hunk 1 above)
Comment (Shahrier-Al-Tanzim) [reply src/controllers/user/user.controllers.js]: ```Okay done```

File: src/views/user/user.dashboard.ejs (hunk 5 above)
Comment (Shahrier-Al-Tanzim) [reply src/views/user/user.dashboard.ejs]: ``` This is an unnecessary comment so I removed it ```

File: src/controllers/user/user.controllers.js (hunk 1 above)
Comment (Talha76) [reply src/controllers/user/user.controllers.js]: Comments not removed yet @Ryuzaki-L-07

File: src/routes/user/user.auth.routes.js [hunk 6]
```diff
@@ -1,24 +1,23 @@
...
-function checkIfAuthenticated(req, res, next) {
-  if (req.isAuthenticated()) {
-    console.log('User is already authenticated');
-    res.redirect('/dashboard');
-  } else {
-    console.log('User is not authenticated');
-    next();
-  }
-}
+// This is the extra layer to check if the user is already authenticated
```
Comment (Talha76) [inline src/routes/user/user.auth.routes.js]: Why unnecessary commented code?

**Pull request merged**